stumpy/core.py

```diff
@@ -659,16 +659,43 @@ def welford_nanstd(a, w=None):
     return np.sqrt(np.clip(welford_nanvar(a, w), a_min=0, a_max=None))
 
 
-def rolling_nanstd(a, w):
+@njit(parallel=True, fastmath={"nsz", "arcp", "contract", "afn", "reassoc"})
+def _rolling_nanstd_1d(a, w):
     """
-    Compute the rolling standard deviation for 1-D and 2-D arrays while ignoring NaNs
-    using a modified version of Welford's algorithm but is much faster than using
-    `np.nanstd` with stride tricks.
+    A Numba JIT-compiled and parallelized function for computing the std of
+    each subsequence with length `w` in `a`, which is a 1D array.
 
-    This a convenience wrapper around `welford_nanstd`.
+    Parameters
+    ----------
+    a : numpy.ndarray
+        The input 1D array
+
+    w : int
+        The rolling window size
+
+    Returns
+    -------
+    out : numpy.ndarray
+        This 1D array has the length of `a.shape[0]-w+1`. `out[i]`
+        contains the std value of `a[i : i + w]`
+    """
+    l = a.shape[0] - w + 1
+    out = np.empty(l, dtype=np.float64)
+    for i in prange(l):
+        out[i] = np.nanstd(a[i : i + w])
+
+    return out
 
-    This essentially replaces:
 
+def rolling_nanstd(a, w, welford=False):
+    """
+    Compute the rolling standard deviation for 1D and 2D arrays while ignoring
```

Comment:
Compute the rolling standard deviation for 1D and 2D arrays while ignoring NaNs.

Reply:
@seanlaw 
Could you please checkout the docstring of this function and your suggestion again? I am trying to see what change I need to apply to the docstring. 
(Did you mean I should have tried to put the word `NaNs` in the first line?)

Reply:
I meant that you should simply shorten it to what I suggested above and remove everything else

Reply:
Ahh.. got it! I will revise it.